docs/contents/dev-docs/testing.md

```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
-    on GitHub. Use this issue as a starting point and if you create a new
-    testing-related issue, mention it from there.
+!!! info 
+     The tests uses the `generate_temp_creds` function from the
+     `scripts/lib/aws.sh` script, executing effectively 
+     ` aws sts assume-role --role-session-arn $AWS_ROLE_ARN --role-session-name $TEMP_ROLE `
+     which fetches temporarily `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`,
+     and an `AWS_SESSION_TOKEN` used in turn to authentication the ACK
+     controller. The duration of the session token is 900 seconds (15 minutes).
+
+Phew that was a lot to set up, but good news: you're almost there.
+
+### Run end-to-end test
 
-## Clean up test runs
+Before you proceed, make sure that you've done the IAM setup in the previous
+step.
 
-To clean up a `kind` Kubernetes cluster, which includes all the
-configuration files created by the script specifically for your test cluster,
-execute:
+!!! warning
+    If you try the following command and you see an error message containing
+    something along the line of `AWS_ROLE_ARN is not defined.` then you know
+    that somewhere in the IAM setup you either left out a step or one of the
+    commands failed.
+
+Now we're finally in the position to execute the end-to-end test:
+
+```
+make kind-cluster SERVICE=$SERVICE
+```
+
+This provisions a Kubernetes cluster using `kind`, builds a container image with
+the ACK service controller, and loads the container image into the `kind` cluster.
+
+It then installs the ACK service controller and related Kubernetes manifests into
+the `kind` cluster using `kustomize build | kubectl apply -f -`.
+
+Then, the above script runs a series of test scripts that call `kubectl`
+and the `aws` CLI tools to verify that custom resources of the type managed by
+the respective ACK service controller is created, updated and deleted
+appropriately (still TODO).
+
+Finally, it will run tests that create resources for the respective service
+and verify if the resource has successfully created. In our example case it
+should create an S3 bucket and then destroy it again, yielding something like
+the following (edited down to the relevant parts):
+
+```
+```
+
+!!! tip "Keeping the test cluster around"
+    By default the script deletes the `kind` cluster. You can prevent this last
+    step from happening by usign `make kind-cluster-preserve SERVICE=$SERVICE`.
+
+### Background
+
+We use [mockery](https://github.com/vektra/mockery) for unit testing.
+You can install it by following the guideline on mockery's GitHub or simply
+by running our handy scirpt at `./scripts/install_mockery.sh` for general
+Linux environments.
+
+
+ We track testing in the umbrella [issue 6](https://github.com/aws/aws-controllers-k8s/issues/6).
+ on GitHub. Use this issue as a starting point and if you create a new
+ testing-related issue, mention it from there.
+
+## Clean up
+
+If you used the `make kind-cluster-preserve` command, the `kind` cluster (Docker
```

Comment:
We should just remove the `kind-cluster-preserve` make target IMHO. The `kind-cluster-functional` target already preserves the test cluster.

Reply:
Once we've cleaned up the Makefile yes. Let's only document what currently exists/works, not aspirational stuff ;)